### ISSUE TITLE
Delay for debugger to attach

### ISSUE THREAD

**Issue by localjo** [hbenl/vscode-firefox-debug#92]
I'm experiencing a delay between launching the debugger in VS Code and having the debugger fully attach. The browser opens, and the page loads, but all of the breakpoints in VS Code are grayed out and the context menu in the browser won't open. After about 30 seconds, the debugger finally finishes attaching, the breakpoints become active, and the context menu starts working in the browser. I'm on a Mac with the latest version of iOS, Firefox and VS Code, using the following config;
```
    {
      "name": "Launch Firefox",
      "type": "firefox",
      "request": "launch",
      "url": "http://localhost:3000",
      "pathMappings": [
        {
          "url": "http://localhost:3000/path/to/my-repo/",
          "path": "${workspaceFolder}/"
        }
      ]
    }
```

Any ideas what might be causing this delay? Has anyone else experienced this? I'm happy to help debug further, but I'm not sure where to start looking, so any hints would be helpful.

**Comment by hbenl** [comment]
If you attach a full [diagnostic log file](https://github.com/hbenl/vscode-firefox-debug#diagnostic-logging) here, I will investigate. I've seen similar delays in the past when working with webpack...

**Comment by localjo** [comment]
With the following config;
```
{
      "name": "Launch Firefox",
      "type": "firefox",
      "request": "launch",
      "url": "http://localhost:3000",
      "pathMappings": [
        {
          "url": "http://localhost:3000/Users/Me/projects/mycompany/myproject-frontend/",
          "path": "${workspaceFolder}/"
        }
      ],
      "skipFiles": ["${workspaceFolder}/node_modules/**", "http?(s):/**"],
      "profile": "default",
      "keepProfileChanges": true,
      "showConsoleCallLocation": true,
      "reAttach": true,
      "reloadOnAttach": true,
      "log": {
        "fileName": "${workspaceFolder}/log.txt",
        "fileLevel": {
          "default": "Debug"
        }
      }
    }
```

I generated this log;

[log.txt](https://github.com/hbenl/vscode-firefox-debug/files/2613481/log.txt)

**Comment by hbenl** [comment]
Thanks. This may be a timing issue in this adapter, could you please try to set `"reloadOnAttach": false` and tell me if you see the same delay when the debugger attaches or when you reload the tab manually afterwards.
Also, the configuration that is responsible for creating the app bundle could be of interest.

**Comment by hbenl** [comment]
Yeah, `create-react-app` uses `webpack` (and configures it so that the URLs of the bundled files look like `http://localhost:3000/absolute/path/to/your/project/file`).
I still couldn't reproduce this, so I added more logging that will hopefully give me an idea what is causing the delay. Please create another log file with the latest version (1.7.2) and attach it here.

**Comment by localjo** [comment]
Sorry for the delay. Here's a log file with version 1.7.3 
[log.txt](https://github.com/hbenl/vscode-firefox-debug/files/2658883/log.txt)

**Comment by hbenl** [comment]
Thanks! It looks like the delay is spent entirely in the [source-map library](https://www.npmjs.com/package/source-map), but I've never heard of similar delays with other debuggers (that also use this library), so it seems I'm doing something wrong when calling it.
I'll have to dig into the source of other debuggers to see how they use this library. Thankfully Christmas vacations are coming up :)

**Comment by digitarald** [comment]
@hbenl with the new source-map behavior in DevTools, is this still an issue?

**Comment by hbenl** [comment]
I never managed to reproduce this, but I had seen some similar delays sporadically. I haven't seen those for a couple of months now, so I guess they were fixed with the recent Firefox debug protocol changes.
@localjo could you check if the delay is still happening?

**Comment by PaulWaldo** [comment]
I am also experiencing this problem.  It takes *75 seconds* to get attached to Firefox.
* Debugger for Firefox Version 2.3.3
* Firefox version 69.0.2
* VS Code version 1.38.1
* MacOS version 10.14.6

My `launch.json` looks like this
```
{
    "version": "0.2.0",
    "configurations": [

        {
            "type": "firefox",
            "request": "launch",
            "reAttach": true,
            "name": "Launch localhost",
            "url": "http://localhost:3000/",
            "webRoot": "${workspaceFolder}/frontend",
            // "skipFiles": ["**/node_modules/**"],
            "log": {
                "fileName": "${workspaceFolder}/log.txt",
                "fileLevel": {
                    "default": "Debug"
                }
            }
        },
        {
            "name": "Python: Django",
            "type": "python",
            "request": "launch",
            "program": "${workspaceFolder}/manage.py",
            "args": [
                "runserver",
                "--noreload"
            ],
            "django": true
        }
    ]
}
```
The log file that gets generates is huge: too large for GitHub to accept.  Please find it here:
https://www.dropbox.com/s/899asql0kwojekm/log.txt.zip?dl=0

Thanks for any suggestions!

**Comment by hbenl** [comment]
Thanks for the log file, it shows that you hit the performance bug that I just fixed a few minutes ago in version 2.3.4.
Please try with that version and tell me if it works for you now.

**Comment by PaulWaldo** [comment]
@hbenl Ask and ye shall receive 😀. Thank you so much, it works quite snappily now!

**Comment by Miguel-Dias-112** [comment]
hello i have the same issue to debbug on my localhost in version 2.9.10
